src/SFA.DAS.AAN.Hub.Data/Repositories/MemberDataCleanupRepository.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.AAN.Hub.Data.Entities;
+
+namespace SFA.DAS.AAN.Hub.Data.Repositories;
+
+public interface IMemberDataCleanupRepository
+{
+    Task<List<Member>> GetWithdrawnOrDeletedMembers();
+    Task<List<Member>> GetRemovedMembers();
+    Task UpdateMemberDetails(Member member, CancellationToken cancellationToken);
+    void DeleteMemberProfile(List<MemberProfile> memberProfiles, CancellationToken cancellationToken);
+    void DeleteMemberPreference(List<MemberPreference> memberPreferences, CancellationToken cancellationToken);
+    void DeleteMemberNotifications(List<Notification> memberNotifications, CancellationToken cancellationToken);
+    void DeleteMemberAudit(List<Audit> memberAudits, CancellationToken cancellationToken);
+    Task DeleteMemberApprentice(Member member, CancellationToken cancellationToken);
+    Task DeleteMemberEmployer(Member member, CancellationToken cancellationToken);
+    Task UpdateMemberFutureAttendance(Member member, CancellationToken cancellationToken);
+}
+
+[ExcludeFromCodeCoverage]
+public class MemberDataCleanupRepository : IMemberDataCleanupRepository
+{
+    private readonly IAanDataContext _context;
+
+    public MemberDataCleanupRepository(IAanDataContext context)
+    {
+        _context = context;
+    }
+    public async Task<List<Member>> GetWithdrawnOrDeletedMembers()
+    {
+        var statuses = new String[] { "Withdrawn", "Deleted" };
+        var query = _context.Members
+            .Where(m => statuses.Contains(m.Status) && m.Email != m.Id.ToString() &&
+                        m.EndDate < DateTime.Today.AddDays(-14))
+            .Include(m => m.MemberPreferences)
+            .Include(m => m.MemberProfiles)
+            .Include(m => m.Notifications)
+            .Include(m => m.Audits);
+
+        return await query.ToListAsync();
+    }
+
+    public async Task<List<Member>> GetRemovedMembers()
+    {
+        var query = _context.Members
+                .Where(m => m.Status == "Removed" && m.Email != m.Id.ToString())
```

Comment:
Can you replace the magic string "Removed" with a constant

Reply:
Magic strings removed and replaced by an enum